### ISSUE TITLE
Build Error Ios

### ISSUE THREAD

**Issue by dnb00** [oracle/pushiomanager-react-native#74]
Im trying to integrate de pushiomanager im facing this issue, I followed the readme and this error keep showing up.

Is the doccsup to date?

im using: expo sdk 51
react native: 0.74
pushiomanager: 7.0

the framework folder

![image](https://github.com/user-attachments/assets/bd8db880-c020-4467-af60-17c86c58bc9c)

![image](https://github.com/user-attachments/assets/f9f63453-731b-4efa-8f01-aa8419cb333c)

**Comment by alexistec** [comment]
same error . any solution?

**Comment by dnb00** [comment]
> same error . any solution?

yeah i fixed this error, the docs are wrong and missing some parts i followed this https://github.com/oracle/pushiomanager-react-native/blob/master/iOS.Example.md when they say to import <PushIOManager/PushIOManager.h> should be <CX_Mobile_SDK/PushIOManager>

**Comment by dnb00** [comment]
using expo you will need to create your own plugin to move files into the correct directory

**Comment by dnb00** [comment]
I did a fork of this library to work with expo eas, because when the expo runs the "install dep" step will break the eas cause dont have cx_mobile_sdk inside framework folder, so i removed this postInstall

**Comment by dnb00** [comment]
This is my expo plugin to create ios, structure... 
`const {
  withDangerousMod,
  withXcodeProject,
  IOSConfig,
} = require('@expo/config-plugins');
const fs = require('fs');
const path = require('path');

module.exports = function withPushIOManager(config) {
  config = withDangerousMod(config, [
    'ios',
    (config) => {
      const iosProjectRoot = config.modRequest.platformProjectRoot;

      const appDelegateHPath = path.join(
        iosProjectRoot,
        config.name.split(' ').join(''),
        'AppDelegate.h'
      );
      const appDelegateMPath = path.join(
        iosProjectRoot,
        config.name.split(' ').join(''),
        'AppDelegate.mm'
      );
      updateAppDelegateH(appDelegateHPath);

      updateAppDelegateM(appDelegateMPath);

      movePushio(config);

      return config;
    },
  ]);
  return withXcodeProject(config, (config) => {
    const xcodeProject = config.modResults;
    const fileName = path.join(
      config.modRequest.projectName,
      'pushio_config.json'
    );

    if (
      !xcodeProject ||
      !xcodeProject.pbxGroupByName ||
      !xcodeProject.getFirstTarget
    ) {
      console.error('❌ Xcode project não está corretamente configurado.');
      return config;
    }

    if (!xcodeProject.hasFile(fileName)) {
      IOSConfig.XcodeUtils.addResourceFileToGroup({
        filepath: fileName,
        groupName: config.modRequest.projectName,
        project: config.modResults,
        isBuildFile: true,
      });
      console.log(`✅ Added ${fileName} to Xcode project`);
    } else {
      console.log(`ℹ️ File ${fileName} already exists in Xcode project`);
    }

    return config;
  });
};

function updateAppDelegateH(filePath) {
  let content = fs.readFileSync(filePath, 'utf-8');

  if (
    !content.includes(
      '#import <react_native_pushiomanager/react-native-pushiomanager-umbrella.h>'
    )
  ) {
    content = content.replace(
      /#import <UIKit\/UIKit.h>/,
      `#import <UIKit/UIKit.h>\n#import <CX_Mobile_SDK/PushIOManager.h>\n#import <UserNotifications/UserNotifications.h>`
    );
  }

  if (!content.includes('UNUserNotificationCenterDelegate')) {
    content = content.replace(
      /@interface AppDelegate : EXAppDelegateWrapper/,
      `@interface AppDelegate : EXAppDelegateWrapper <UIApplicationDelegate, UNUserNotificationCenterDelegate>`
    );
  }

  fs.writeFileSync(filePath, content, 'utf-8');
  console.log('Updated AppDelegate.h for PushIOManager.');
}

function updateAppDelegateM(filePath) {
  const customAppDelegatePath = path.join('plugins/custom', 'AppDelegate.mm');

  // Replace the AppDelegate.m file
  fs.copyFileSync(customAppDelegatePath, filePath);
}

function movePushio(config) {
  const projectRoot = config.modRequest.projectRoot;
  const fileName =
    process.env.APP_VARIANT === 'production'
      ? 'pushio_config_ios_prod.json'
      : 'pushio_config_ios_debug.json';
  const sourceFilePath = path.join(projectRoot, 'sdk', fileName);

  // Path to the destination folder: ios/<AppName>/pushio_config
  const iosFolderPath = path.join(
    config.modRequest.platformProjectRoot,
    config.modRequest.projectName
  );
  const destinationFilePath = path.join(iosFolderPath, 'pushio_config.json');

  // Check if the source file exists
  if (fs.existsSync(sourceFilePath)) {
    // Move and rename the file
    fs.cpSync(sourceFilePath, destinationFilePath, { recursive: true });
    console.log(
      `Moved and renamed ${sourceFilePath} to ${destinationFilePath}`
    );
  } else {
    console.warn(`File "${sourceFilePath}" not found at ${sourceFilePath}`);
    process.exit(1);
  }
}`

**Comment by joaovitorleffa** [comment]
> > same error . any solution?
> 
> yeah i fixed this error, the docs are wrong and missing some parts i followed this https://github.com/oracle/pushiomanager-react-native/blob/master/iOS.Example.md when they say to import <PushIOManager/PushIOManager.h> should be <CX_Mobile_SDK/PushIOManager>

Great! Thanks for sharing. However, in my case, I had to use <CX_Mobile_SDK/PushIOManager.h> instead of <CX_Mobile_SDK/PushIOManager> for it to work correctly.
